### PR TITLE
test: fix flaky tests `Dapp viewed Event...` `Permissions Page...` and `Connections Page...` due to overlaying popup

### LINKED ISSUE #24575: Flaky Test: "Dapp viewed Event @no-mmi is sent when reconnect to  a dapp that has been connected before"
Context:

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/80401/workflows/55c49ded-58c2-487a-8ab2-afb21f8f8a2e/jobs/2836310/tests

### PR DESCRIPTION
**Pull request by seaona**
## **Description**
This is a PR for fixing the flaky tests:
- Dapp viewed Event @no-mmi is sent when reconnect to a dapp that has been connected before
- Permissions Page should redirect users to connections page when users click on connected permission
- Connections page should disconnect when click on Disconnect button in connections page

The problem seems to be that we are trying to click an element, but there is the previous notifcation element on top and has not disappeared yet at the moment we want to perform the click operation.

We can see this in the artifacts of the failure, where the blue notification popup is still there, despite having already clicked 'Got it'.
This is also confirmed by the error message in the logs:

```
[driver] Called 'clickElement' with arguments [{"text":"Got it","tag":"button"}]
[driver] Called 'clickElement' with arguments [{"text":"127.0.0.1:8080","tag":"p"}]
Failure on testcase: 'Dapp viewed Event @no-mmi is sent when reconnect to a dapp that has been connected before', for more information see the artifacts tab in CI
ElementClickInterceptedError: Element <p class="mm-box mm-text mm-text--body-md mm-text--elllipsis mm-text--text-align-left mm-box--color-text-default">
 is not clickable at point (444,92) because another element <h4 class="mm-box mm-text mm-text--heading-sm mm-text--text-align-center mm-box--width-full mm-box--color-info-inverse"> obscures it
```

The solution here is to wait for the element to disappear. Since this seems a case that can happen more often, a new driver function has been introduced, which basically clicks an element and waits until it disappears. This can be suitable for when accepting notifications/popup that whenever we accept them, those should disappear before we proceed with the next action.

```
  async clickElementAndWaitToDisappear(rawLocator) {
    const element = await this.findClickableElement(rawLocator);
    await element.click();
    await element.waitForElementState('hidden');
  }
 ```

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24575 and https://github.com/MetaMask/metamask-extension/issues/24597

## **Manual testing steps**

1. locally: run the test locally several times ` yarn test:e2e:single test/e2e/tests/metrics/dapp-viewed.spec.js --browser=chrome --leave-running=true`
2. ci: Check ci and see is not failing

## **Screenshots/Recordings**
Artifacts and logs from ci that helped in the investigation:

image

Screenshot from 2024-05-17 11-27-45

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/webdriver/driver.js [hunk 1]
```diff
@@ -373,6 +373,12 @@ class Driver {
     await element.click();
   }
 
+  async clickElementAndWaitToDisappear(rawLocator) {
+    const element = await this.findClickableElement(rawLocator);
+    await element.click();
+    await element.waitForElementState('hidden');
```
Comment (legobeat) [inline test/e2e/webdriver/driver.js:379]: Would it make sense to set an explicit timeout here of the maximum time we expect the notification element to be visible in this step?

Comment (seaona) [reply test/e2e/webdriver/driver.js]: absolutely @legobeat :bowing_woman: :heart:  I've pushed a new commit to address that

**Review by legobeat — DISMISSED** [review]
LGTM!

**Comment by seaona** [comment]
@legobeat there were more instances of this type of flakyiness, so I've decided to add the fixes in this same PR. Apologies for resetting the review, I just saw it now in this ticket 
:pray: :bowing_woman: 
https://github.com/MetaMask/metamask-extension/issues/24597

**Review by legobeat — APPROVED** [review]

**Review by chloeYue — APPROVED** [review]
LGTM! Thanks !

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**